corehq/apps/reports/tests/test_application_status_report.py

```diff
@@ -42,15 +51,47 @@ def setUpClass(cls):
         cls.addClassCleanup(cls.admin_user.delete, cls.domain_name_a, deleted_by=None)
 
         user1 = CommCareUser.create(cls.domain_name_a, 'mobile_worker1', '***', None, None)
+        # app_b being in different domain will cause user1 to be skipped from app status report
+        # via AppInDifferentDomainException when viewing for all apps
         user1.reporting_metadata = cls._reporting_metadata_multiple_apps(cls.app_a._id, cls.app_b._id)
-        user2 = CommCareUser.create(cls.domain_name_a, 'mobile_worker2', '***', None, None)
+        user2 = CommCareUser.create(cls.domain_name_a, 'mobile_worker2', '***', None, None,
+                                    location=domain_a_city)
         user2.reporting_metadata = cls._reporting_metadata_single_app(cls.app_a._id)
 
-        for user in [user1, user2]:
+        user3 = CommCareUser.create(cls.domain_name_a, 'mobile_worker3', '***', None, None,
+                                    location=domain_a_state)
+        user3.reporting_metadata = cls._reporting_metadata_single_app(cls.app_a._id, app_build_version=10)
+        user4 = CommCareUser.create(cls.domain_name_a, 'mobile_worker4', '***', None, None,
+                                    location=domain_a_country)
+        user4.reporting_metadata = cls._reporting_metadata_single_app(cls.app_a._id, app_build_version=15)
+
+        for user in [user1, user2, user3, user4]:
             user.save()
             user_adapter.index(user, refresh=True)
             cls.addClassCleanup(user.delete, cls.domain_name_a, deleted_by=None)
 
+    @classmethod
+    def _setup_locations(cls, domain):
+        loc_type_country = LocationType.objects.create(domain=domain, name='Country')
+        loc_type_state = LocationType.objects.create(domain=domain, name='State', code=LOCATION_TYPE_STATE,
+                                                     parent_type=loc_type_country)
+        loc_type_city = LocationType.objects.create(domain=domain, name='City', code=LOCATION_TYPE_CITY,
+                                                    parent_type=loc_type_state)
+        loc_country = make_location(site_code='india', name='India', domain=domain,
+                                    location_type=loc_type_country.name)
+        loc_country.save()
+        loc_state = make_location(site_code='mh', name='Maharashtra', domain=domain,
+                                  location_type=loc_type_state.name, parent=loc_country)
+        loc_state.save()
+        loc_city = make_location(site_code='bmb', name='Bombay', domain=domain,
+                                 location_type=loc_type_city.name, parent=loc_state)
+        loc_city.save()
+
+        # cleans up the full hierarchy
+        cls.addClassCleanup(loc_country.location_type.delete)
```

Comment:
Are these in SQL? If yes, they will be cleaned up automatically by transaction rollback in class cleanup.